client/client.go

```diff
@@ -462,6 +462,46 @@ func (r *NotaryRepository) GetTargetByName(name string, roles ...string) (*Targe
 
 }
 
+// CheckTargetByName returns true if all passed-in roles have signed the specified target with the same hash.
+// If no roles are passed, it uses the targets role.  Note that child roles are considered if
+// we can't find the target in the specified role.
+func (r *NotaryRepository) CheckTargetByName(name string, roles ...string) (bool, error) {
+	if err := r.Update(false); err != nil {
+		return false, err
+	}
+
+	if len(roles) == 0 {
```

Comment:
Originally I had this function only check the specified role without looking in child roles so it was clearer, do you think that's better?

I'd be ok with filtering out roles (though it would add some complexity), but I'm not sure if we want to make the case you described above as an error case -- it seems that if a target is signed by `targets/releases`, it's been delegated to sign by the `targets` role anyway?